### ISSUE TITLE
onEnter function in Route does not get called

### ISSUE THREAD

**Issue by ateufel** [rackt/react-router#1752]
Not sure if this an issue on my end or if the examples are just outdated. I got the following simple code, nevermind the missing parameters of the "requireAuth" function, it´s just for testing:
```javascript
function requireAuth () {
	console.log('test');
}
var routes = (
	<Route handler={App} path="/">
		<DefaultRoute handler={Login} />
		<Route path="overview" handler={Overview} onEnter={requireAuth} />
	</Route>
);
Router.run(routes, Router.HashLocation, function (Root, state) {
	React.render(<Root />, document.getElementById('react'));
});
```
The requireAuth function just never gets called, and i have no clue why. I started debugging "Route.js" from the router, and onEnter is always undefined. Is there anything i am missing or does this just not work anymore? Another solution would be to check for authorization in the Router callback and add an option to the Route, but i don´t think you can use custom options:

```javascript
<Route path="overview" handler={Overview} needsAuth={true} />
```

Or is there another recommended way to implement this?

**Comment by taurose** [comment]
`<Route onEnter>`, and accessing arbitrary route props (like your `needsAuth`) is not supported in 0.13.3, but will likely be in 1.0.

Instead of `onEnter` you can use `willTransitionTo` as a static method on your handlers. See [docs](http://rackt.github.io/react-router/#Route Handler) and [example](https://github.com/rackt/react-router/blob/0.13.x/examples/auth-flow/app.js).

**Comment by ateufel** [comment]
oh, i thought it´s available already, because i saw some code examples using onEnter...i´d love to avoid putting the auth check in the handler, but i guess that´s the way to go. or would you recommend using the current master branch instead? it looks like onEnter is in there already: https://github.com/rackt/react-router/blob/master/examples/auth-flow/app.js

**Comment by taurose** [comment]
If you're only concerned with declaring the auth requirements close to your route config, you could do it like in the example, and use higher order components.

```js
<Route handler={requireAuth(Component)} />
```

As for using master branch, you can try out the beta versions which have their own git tags, docs and are published to npm (`npm i react-router@1.0.0-beta3`). There are still some open issues, and the master branch has already moved on a bit since the latest beta release. So I'm not sure what to recommend. Depends on how adventurous you are I guess ;)

**Comment by ateufel** [comment]
well, i am very adventurous, but not that much when it comes to projects for a customer :)
